parcel-bundler/parcel
# Format Rust in precommit and lint+format scripts

Depends on #6200.

This:
* Runs `rustfmt` on staged files in a precommit hook just as we do JS with prettier. 
* Uses `cargo fmt` in `lint` and `format` yarn scripts.
* Formats existing code.

Notes:
* Also updates husky to version 6 using typicode/husky-4-to-6, which uses files in a directory instead of package.json config.
* Set the `tab_spaces` configuration to 2 spaces to reduce churn with existing code. The Rust community and rustfmt seems to have settled on 4 spaces a default. Happy to discuss and change.

Test Plan: 
* Use rust-analyzer VSCode extension with automatic formatting and verify no diff churn on save.
* Make a change to a Rust file with unnecessary formatting and save without formatting the file. Commit the file and verify the unnecessary formatting is removed.
* Make a change to a Rust file with unnecessary formatting and run `yarn lint` and verify it fails. Run `yarn format` and verify the formatting is removed.
